### PR TITLE
Additional API definitions for backup

### PR DESCRIPTION
**Pull request by DavidePrincipi**
Refs https://github.com/NethServer/ns8-scratchpad/pull/137

- add-backup-repository
- list-backup-repositories
- add-backup
- list-backups

### CODE REVIEW

File: core/imageroot/var/lib/nethserver/cluster/actions/add-backup/10validate [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+errors = []
+
+request = json.load(sys.stdin)
+rdb = agent.redis_connect(privileged=False)
+
+agent.set_weight(os.path.basename(__file__), 0) # Validation step, no task progress at all
+
+repository = request['repository']
+
+# check for duplicate names
```
Comment (gsanchietti) [inline core/imageroot/var/lib/nethserver/cluster/actions/add-backup/10validate:39]: It seems to me that the comment is lying here :)

File: core/imageroot/var/lib/nethserver/cluster/actions/list-backup-repositories/validate-output.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                "password": "d59a90ec7ad2b2967257a7a308c82c96ac006efd138254bc1e58c8ea07c18400",
+                "parameters": {
+                    "b2_account_id": "xxxxxxxxxxxxxx",
+                    "b2_account_key": "yyyyyyyyyyyyyyyyyyyyyy"
+                }
+            }
+        ]
+    ],
+    "type": "array",
+    "$defs": {}
```
Comment (gsanchietti) [inline core/imageroot/var/lib/nethserver/cluster/actions/list-backup-repositories/validate-output.json:21]: If we do not want to create an extended definition, can we remove this line?

File: core/imageroot/var/lib/nethserver/cluster/actions/list-backups/validate-output.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                    }
+                ]
+            }
+        ]
+    ],
+    "type": "array",
+    "items": {
+        "type": "object"
+    },
+    "$defs": {}
```
Comment (gsanchietti) [inline core/imageroot/var/lib/nethserver/cluster/actions/list-backups/validate-output.json:29]: I'd remove also this one, if unused.

**Review by gsanchietti — APPROVED** [review]
Please bear in mind that merging this one will break #137

**Comment by DavidePrincipi** [comment]
> Please bear in mind that merging this one will break #137

AFAIK it is "compatible" with it, though it aims to change the module APIs

**Pull request merged**